### PR TITLE
Fix deleting a disk from VM

### PR DESCRIPTION
**Pull request by borod108**
Removing a disk was not handled correctly

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1666593

### CODE REVIEW

**Comment by borod108** [comment]
@agrare @masayag what do you think?

File: app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb [hunk 1]
```diff
@@ -272,7 +272,6 @@ def vm_reconfigure(vm, options = {})
         # Retrieve the current representation of the virtual machine:
         # TODO: no need to retreive vm here, only if memory is updated
         vm = vm_service.get
-
```
Comment (masayag) [inline app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: seems this space line is required

Comment (borod108) [reply app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: added it back

File: app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb [hunk 2]
```diff
@@ -824,8 +824,18 @@ def prepare_vm_disk_attachment(disk_spec, storage_spec)
...
       attachments_service = vm_service.disk_attachments_service
+
+      disk_attachment_by_disk_name_hash = attachments_service.list.collect do |attachment_service_obj|
+        attachment_service = attachments_service.attachment_service(attachment_service_obj.id)
+        [vm_service.connection.follow_link(attachment_service_obj.disk).name, attachment_service]
+      end.to_h
+
       disk_specs.each do |disk_spec|
-        attachment_service = attachments_service.attachment_service(disk_spec['disk_name'])
+        disk_spec = disk_spec.with_indifferent_access
```
Comment (agrare) [inline app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: @borod108 just curious, why do you need with_indifferent_access here?  Is the `'disk_name'` key not always a string ?

Comment (borod108) [reply app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: When the request comes from api it is, at least sometimes, a symbol (this was one of the things that went wrong here)

Comment (agrare) [reply app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: Interesting ok thanks

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 292239cc99e49cebe0b84e68ef85c36fa24b3e70
Author: Adam Grare <agrare@redhat.com>
Date:   Mon Mar 11 10:03:03 2019 -0400

    Merge pull request #348 from borod108/bugs/fix1666593
    
    Fix deleting a disk from VM
    
    (cherry picked from commit c4063dd78b9c0fac8131ae51d68d92fb994de0be)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1687846
```

**Pull request merged**